packages/intrepid2/unit-test/Shared/ArrayTools/CMakeLists.txt

```diff
@@ -1,15 +1,44 @@
 INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR})
 
+SET(Intrepid2_TEST_ETI_FILE "test_01;test_02;test_03;test_04;test_05")
+
+SET(Intrepid2_TEST_ETI_DEVICE_NAME "")
 IF(Kokkos_ENABLE_SERIAL)
-  ADD_SUBDIRECTORY(Serial)
+  LIST(APPEND Intrepid2_TEST_ETI_DEVICE_NAME "Serial")
 ENDIF()
-
 IF(Kokkos_ENABLE_OPENMP)
-  ADD_SUBDIRECTORY(OpenMP)
+  LIST(APPEND Intrepid2_TEST_ETI_DEVICE_NAME "OpenMP")
 ENDIF()
-
 IF(Kokkos_ENABLE_CUDA)
-  ADD_SUBDIRECTORY(Cuda)
+  LIST(APPEND Intrepid2_TEST_ETI_DEVICE_NAME "CUDA")
 ENDIF()
+IF(Kokkos_ENABLE_HIP)
+  LIST(APPEND Intrepid2_TEST_ETI_DEVICE_NAME "HIP")
+ENDIF()
+
+FOREACH(ETI_DEVICE_NAME IN LISTS Intrepid2_TEST_ETI_DEVICE_NAME)
+  FOREACH(ETI_FILE IN LISTS Intrepid2_TEST_ETI_FILE)
```

Comment:
This ETI approach for testing looks like a major improvement for the test build system.  I have not yet understood details of how it works, but I love the fact that we can replace the per-architecture CMakeLists.txt.